### PR TITLE
Updating the azure/login version to latest (v2) and adding a validati…

### PR DESCRIPTION
**Pull request by emvaldes**
This PR is attempting to:

- Update the existing azure/login action to the latest version (currently v2).

**This is not a fix for a currently exploitable issue.**

Test Steps:
1. *Testing the existing **Service Principal** credentials are valid*

## Changes
- *Updating **Service Principal** credentials for the **PRIME-ReportStream** application*

### CODE REVIEW

File: .github/actions/build-vars/action.yml [hunk 1]
```diff
@@ -234,11 +234,18 @@ runs:
           echo "has_frontend_change=${{ steps.filter.outputs.frontend_react }}" >> $GITHUB_OUTPUT
         fi
 
-    - uses: azure/login@a65d910e8af852a8061c627c456678983e180302
+    - uses: azure/login@v2
```
Comment (jack-h-wang) [inline .github/actions/build-vars/action.yml:237]: For reference, this project has traditionally pinned dependencies by hash for the reasons outlined here: https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

Comment (emvaldes) [reply .github/actions/build-vars/action.yml:237]: Thank you for pointing that out but at some point we need to catch up with the latest release for these GitHub Actions. Nevertheless, we are just attempting to determine what if it works but not forcing this to be a change merged into the main branch yet.

Comment (devopsmatt) [reply .github/actions/build-vars/action.yml:237]: This action is from a [verified source](https://github.com/marketplace/actions/azure-login) (Azure) and has defined CODEOWNERS, so I am not that concerned with sha pinning at the moment.

**Comment by emvaldes** [comment]
``` console
Run azure/login@v2
  with:
    creds: {"clientId":"***","clientSecret":"***","subscriptionId":"***","tenantId":"***"}
    enable-AzPSSession: false
    environment: azurecloud
    allow-no-subscriptions: false
    audience: api://AzureADTokenExchange
    auth-type: SERVICE_PRINCIPAL
  env:
    AZURE_CREDENTIALS: {"clientId":"***","clientSecret":"***","subscriptionId":"***","tenantId":"***"}
##[debug]Reading creds in JSON...
::add-mask::***
::add-mask::***
Running Azure CLI Login.
##[debug]Azure CLI path: /usr/bin/az
##[debug]Azure CLI version used:
##[debug]{
##[debug]  "azure-cli": "2.67.0",
##[debug]  "azure-cli-core": "2.67.0",
##[debug]  "azure-cli-telemetry": "1.1.0",
##[debug]  "extensions": {
##[debug]    "azure-devops": "1.0.1"
##[debug]  }
##[debug]}
##[debug]
/usr/bin/az cloud set -n azurecloud
Done setting cloud: "azurecloud"
Note: Azure/login action also supports OIDC login mechanism. Refer https://github.com/azure/login#configure-a-service-principal-with-a-federated-credential-to-use-oidc-based-authentication for more details.
Attempting Azure CLI login by using service principal with secret...
Subscription is set successfully.
Azure CLI login succeeds by using service principal with secret.
##[debug]Node Action run completed with exit code 0
##[debug]Finished: run
##[debug]Evaluating: inputs.sp-creds
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'sp-creds'
##[debug]=> '{"clientId":"***","clientSecret":"***","subscriptionId":"***","tenantId":"***"}'
##[debug]Result: '{"clientId":"***","clientSecret":"***","subscriptionId":"***","tenantId":"***"}'
##[debug]Evaluating condition for step: 'run'
##[debug]Evaluating: (success() && (inputs.tf-auth == 'true') && inputs.sp-creds)
##[debug]Evaluating And:
##[debug]..Evaluating success:
##[debug]..=> true
##[debug]..Evaluating Equal:
##[debug]....Evaluating Index:
##[debug]......Evaluating inputs:
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'tf-auth'
##[debug]....=> 'false'
##[debug]....Evaluating String:
##[debug]....=> 'true'
##[debug]..=> false
##[debug]=> false
##[debug]Expanded: (true && ('false' == 'true') && inputs['sp-creds'])
##[debug]Result: false
##[debug]Finishing: Connect to VPN and login to Azure
```

**Review by emvaldes — COMMENTED** [review]
The goal is to dynamically generate AZURE_CREDENTIALS into a JSON object to replace the existing method of hardcoding all parameters.
Removing the hardcoding of these azure/login hashed commits in favor of version specific from those vendors considered to be trusted.

**Review by devopsmatt — APPROVED** [review]
LGTM, good work!

**Pull request merged**